app/AI4Green4Students/Services/PredictionService.cs

```diff
@@ -10,34 +10,35 @@ namespace AI4Green4Students.Services;
 
 public class PredictionService
 {
-  private readonly WorkerOptions _worker;
+  private readonly PredictorOptions _predictor;
 
-  public PredictionService(IOptions<WorkerOptions> worker) => _worker = worker.Value;
+  public PredictionService(IOptions<PredictorOptions> predictor) => _predictor = predictor.Value;
 
   /// <summary>
-  /// Predict products based on reaction smiles.
+  /// Forward prediction based on reaction smiles.
   /// </summary>
   /// <param name="smiles">Reaction smiles.</param>
-  /// <param name="firstN">Number of products to return.</param>
-  /// <returns>Predicted products.</returns>
-  public async Task<List<PredictedProductModel>> PredictProducts(string smiles, int firstN = 5)
+  /// <returns>Results.</returns>
+  public async Task<List<ForwardPredictionModel>> ForwardPrediction(string smiles)
   {
-    var url = _worker.ApiUrl.TrimEnd('/') + "/api"
-      .AppendPathSegment("predict-products")
-      .SetQueryParams(new
-      {
-        smiles, firstN
-      });
+    var url = _predictor.ForwardPredictionApiUrl.TrimEnd('/') + "/api"
+      .AppendPathSegment("predict");
 
-    var response = await url.WithHeader("x-functions-key", _worker.ApiKey).GetStringAsync();
-    var data = JsonSerializer.Deserialize<List<PredictedProductDataModel>>(response, DefaultJsonOptions.Serializer);
+    var requestBody = new { smiles };
 
-    return data?.Select(x => new PredictedProductModel(
+    var response = await url
+      .WithHeader("X-API-Key", _predictor.ForwardPredictionApiKey)
+      .PostJsonAsync(requestBody)
+      .ReceiveString();
+
+    var data = JsonSerializer.Deserialize<ForwardPredictionResultDataModel>(response, DefaultJsonOptions.Serializer);
+
+    return data?.Result.Select(x => new ForwardPredictionModel(
       x.Product,
       Math.Round(x.Score, 2),
       x.ReactionImage,
       x.IupacName,
       x.Synonyms
-    )).ToList() ?? new List<PredictedProductModel>();
+    )).ToList() ?? new List<ForwardPredictionModel>();
```

Comment:
is the `??` bit catching if `data` comes back with nothing?

Reply:
yep, if no data or deserialization fails then this return empty list